src/main/java/CliUI.java

```diff
@@ -0,0 +1,113 @@
+import java.util.Scanner;
+
+class CliUI {
+    private SnippetManager snippetManager;
+    private boolean isOpen;
+    private Scanner keyboard;
+
+    CliUI(SnippetManager snippetManager) {
+        this.snippetManager = snippetManager;
+        isOpen = true;
+        keyboard = new Scanner(System.in);
+    }
+
+    void uiLoop() {
+        while (isOpen) {
+            displayMenu();
+            getAndExecuteCommand();
+        }
+    }
+
+    private void displayMenu() {
+        System.out.println("Menu:\n1. View all snippets\n2. Create a snippet\n3. Delete a snippet\n4. Edit a snippet\n5. Filter\n6. Quit");
+    }
+
+    private void getAndExecuteCommand() {
+        int input = keyboard.nextInt();
+
+        switch (input) {
```

Comment:
Good use of the cases and keeping the code clear by putting functions as the cases.